src/pyo3_extensions/mod.rs

```diff
@@ -196,84 +183,19 @@ impl PartialEq for TdPyAny {
     }
 }
 
-/// Newtype over a string representing the "routing key" for stateful
-/// operators.
-///
-/// We use a string here rather than wrapping any Python type because
-/// the routing key interfaces with a lot of Bytewax and Timely code
-/// which puts requirements on it: it has to be hashable, have
-/// equality, debug printable, and is serde-able and we can't
-/// guarantee those things are correct on any arbitrary Python type.
-///
-/// Yes, we lose a little bit of flexibility, but it makes usage more
-/// convenient.
-///
-/// You'll mostly interface with this via [`extract_state_pair`] and
-/// [`wrap_state_pair`].
-#[derive(
-    Debug, Clone, PartialEq, Eq, Hash, PartialOrd, Ord, Serialize, Deserialize, FromPyObject,
-)]
-pub(crate) struct StateKey(String);
```

Comment:
Moved into `crate::recovery` since it only exists to support the recovery data model. Then we can keep all the SQLx stuff isolated in there too.